samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -3,8 +3,62 @@
 e.g. linking layers to functions
 """
 
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
 import re
-from typing import List
+
+
+@dataclass
+class ConstantValue:
+    value: Any
+
+
+@dataclass
+class References:
+    value: List[str]
+
+
+Expression = Union[ConstantValue, References]
+
+
+@dataclass
+class ResolvedReference:
+    value: str
+    module_address: str
+
+
+@dataclass
+class TFModule:
+    # full path to the module, including parent modules
+    full_address: Optional[str]
+    parent_module: Optional["TFModule"]
+    variables: Dict[str, Expression]
```

Comment:
I like the change to use Expression instead of the old TFValue class. It looks cleaner. Just let Daniel and Lucas know about this change as it may affect their implementation